src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -1798,8 +1803,7 @@ public void unlock(final UnlockKey unlockKey) throws DockerException, Interrupte
   @Override
   public ServiceCreateResponse createService(ServiceSpec spec)
       throws DockerException, InterruptedException {
-
-    return createService(spec, registryAuth);
+    return createService(spec, registryAuthSupplier.authFor(""));
```

Comment:
it is not documented in the authFor method that you can pass special String values to get special RegistryAuth instances - why not just have a 2nd method in the interface for this use case?